test/unit/parser.test.js

```diff
@@ -5,34 +5,48 @@ describe('Standard Markdown', () => {
-    expect(output).toEqual("This is some text with [a link]\n");
+    expect(output).toEqual("This is some text with [a link][0]\n");
   });
 
   it('redacts image urls', () => {
     const input = "This is some text with ![an image](http://example.com/img.jpg)";
     const output = parser.sourceToRedacted(input);
-    expect(output).toEqual("This is some text with ![an image]\n");
+    expect(output).toEqual("This is some text with [an image][0]\n");
   });
 
   it('can merge a source MDAST with a redacted MDAST', () => {
     const source = "This is some text with [a link](http://example.com/)";
-    const redacted = "Ceci est un texte avec [un lien]";
+    const redacted = "Ceci est un texte avec [un lien][0]";
     const output = parser.sourceAndRedactedToHtml(source, redacted);
     expect(output).toEqual("<p>Ceci est un texte avec <a href=\"http://example.com/\">un lien</a></p>\n");
   });
 
+  it('can merge a source MDAST with a redacted MDAST', () => {
+    const source = "This is some text with ![an image](http://example.com/img.jpg)";
+    const redacted = "Ceci est un texte avec [une image][0]";
+    const output = parser.sourceAndRedactedToHtml(source, redacted);
+    expect(output).toEqual("<p>Ceci est un texte avec <img src=\"http://example.com/img.jpg\" alt=\"une image\"></p>\n");
+  });
+
   it('can differentiate between multiple redacted links', () => {
     const source = "This is some text with [a link](http://first.com) and ![an image](http://second.com/img.jpg).\n\nAnd also a second paragraph with [another link](http://third.com)";
-    const redacted = "C'est du texte avec [un lien] et ![une image].\n\nEt aussi un deuxième paragraphe avec [un autre lien]";
+    const redacted = "C'est du texte avec [un lien][0] et [une image][1].\n\nEt aussi un deuxième paragraphe avec [un autre lien][2]";
     const output = parser.sourceAndRedactedToHtml(source, redacted);
     expect(output).toEqual("<p>C'est du texte avec <a href=\"http://first.com\">un lien</a> et <img src=\"http://second.com/img.jpg\" alt=\"une image\">.</p>\n<p>Et aussi un deuxième paragraphe avec <a href=\"http://third.com\">un autre lien</a></p>\n");
   });
 
-  it('if the redacted text removes a url, it will blindly merge however many urls there are', () => {
+  it('can handle reordering of redacted links', () => {
+    const source = "This is some text with [a link](http://first.com) and ![an image](http://second.com/img.jpg).\n\nAnd also a second paragraph with [another link](http://third.com)";
+    const redacted = "C'est du texte avec [un lien][0] et [une image][2].\n\nEt aussi un deuxième paragraphe avec [un autre lien][1]";
+    const output = parser.sourceAndRedactedToHtml(source, redacted);
+    expect(output).toEqual("<p>C'est du texte avec <a href=\"http://first.com\">un lien</a> et <a href=\"http://third.com\">une image</a>.</p>\n<p>Et aussi un deuxième paragraphe avec <img src=\"http://second.com/img.jpg\" alt=\"un autre lien\"></p>\n");
```

Comment:
It's a good test, but a contrived example that looks like it results in incorrect output.  An adjective somewhere in there would demonstrate how language can necessitate reordering.

```js
const source = "The [cold](http://first.com) [cat](http://second.com)."
const redacted = "Le [chat][1] [froid][0]."
```

Reply:
Much better! Yeah, this test was the result of me updating the "we do not account for ordering" test to reflect the new functionality without actually thinking about the test content itself.